plugin.py

```diff
@@ -77,5 +101,25 @@ def Run(self):
 else:
     theApp = wx.GetApp()
 
+
+# run functon inside gui-thread-safe context, requires wx.App
+def safe_invoke(function):
+    wx.CallAfter(function)
+
+
+# verify required pcbnew api is present
+def has_pcbnew_api():
+    return hasattr(pcbnew, 'ExportSpecctraDSN') and hasattr(pcbnew, 'ImportSpecctraSES')
+
+
+# report error to the user gui
+def show_error(message):
+    caption = "kicad_freerouting-plugin"
```

Comment:
nit: "Freerouting plugin"
as user string it doesn't need to avoid spaces

Reply:
applied